pennylane/pulse/transmon.py

```diff
@@ -0,0 +1,193 @@
+
+    TODO: resource / source for good values.
+
+    .. seealso::
+
+        :func:`~.drive`
+
+    Args:
+        connections (list[tuple(int)]): List of connections ``(i, j)`` between qubits i and j.
+        omega (Union[float, list[float]]): List of dressed qubit frequencies in GHz.
+            When passing a single float all qubits are assumed to have that same frequency.
+        g (Union[float, list[float]]): List of coupling strengths in GHz. Needs to match the length of ``connections``.
+            When passing a single float all connections are assumed to have the same strength.
+        delta (Union[float, list[float]]): List of anharmonicities in GHz. Ignored when ``d=2``.
+            When passing a single float all qubits are assumed to have that same anharmonicity.
+        wires (list): Optional, defaults to the unique wires in ``connections`` when set to ``None``. When passing explicit ``wires``,
+            needs to at least contain all unique wires in ``connections`` and can be used to initiate additional, unconnected qubits.
+        d (int): Local Hilbert space dimension. Defaults to ``d=2`` and is currently the only supported value.
+
+    Returns:
+        HardwareHamiltonian: a :class:`~.ParametrizedHamiltonian` representing the transmon interaction
+
+    **Example**
+
+    We can set up the transmon interaction Hamiltonian with uniform coefficients by passing ``float`` values.
+
+    .. code-block::
+
+        connections = [[0, 1], [1, 3], [2, 1], [4, 5]]
+        H = qml.pulse.transmon_interaction(connections, omega=0.5, g=1.)
+
+    The resulting :class:`~.ParametrizedHamiltonian` consists of ``4`` coupling terms and ``6`` qubit terms.
+
+    >>> print(H)
+    ParametrizedHamiltonian: terms=10
+
+    """
+    if d != 2:
+        raise NotImplementedError(
+            "Currently only supporting qubits. Qutrits and qudits are planned in the future."
+        )
+
+    if wires is not None and not all(i in wires for i in qml.math.unique(connections)):
+        raise ValueError(
+            f"There are wires in connections {connections} that are not in the provided wires {wires}"
+        )
+
+    wires = wires or qml.math.unique(connections)
+    n_wires = len(wires)
+
+    # Prepare coefficients
+    # TODO: make coefficients callable / trainable. Currently not supported
+    if qml.math.ndim(omega) == 0:
+        omega = [omega] * n_wires
+    if len(omega) != n_wires:
+        raise ValueError(
+            f"Number of qubit frequencies omega = {omega} does not match the provided wires = {wires}"
+        )
+
+    if qml.math.ndim(g) == 0:
+        g = [g] * len(connections)
+    if len(g) != len(connections):
+        raise ValueError(
+            f"Number of coupling terms {g} does not match the provided connections = {connections}"
+        )
+
+    # qubit term
+    coeffs = list(omega)
+    observables = [ad(i, d) @ a(i, d) for i in wires]
```

Comment:
So, if we use the unified `drive` term (with the detuning term), this is already included in drive, and should not be included here in the interaction term. I.e. the \Delta term in the drive is a combination of this and the applied drive. Having this here breaks the drive term.